### ISSUE TITLE
Cannot reproduce the results on LRS3 test set

### ISSUE THREAD

**Issue by joyolee** [JeongHun0716/MMS-LLaMA#1]
Dear authors,

thank you very much for the great work. We downloaded the pre-trained models and tried reproducing the results on the LRS3 dataset.
With clean speech, we have larger WERs 35.12% on the LRS3 test set.

The following are our configurations for decoding:

```
ROOT=$(pwd)
SRC_PTH=$ROOT/src
MODEL_PATH=$ROOT/pretrained_models/mms_llama/1759h/[ckpt.pt](http://ckpt.pt/)

LLM_PATH=meta-llama/Llama-3.2-3B

OUT_PATH=$ROOT/results/

PYTHONPATH=$ROOT/fairseq
python3 -B $SRC_PTH/[eval.py](http://eval.py/) --config-dir ${SRC_PTH}/conf --config-name s2s_decode
dataset.gen_subset=test
common.user_dir=${SRC_PTH}
generation.beam=5
generation.temperature=0.3
override.llm_path=${LLM_PATH}
override.modalities=['video','audio']
common_eval.path=${MODEL_PATH}
common_eval.results_path=${OUT_PATH}
override.label_dir=$ROOT/manifest/433h
[override.data](http://override.data/)=$ROOT/manifest/433h
override.noise_wav=$ROOT/noise/babble_noise.wav
override.noise_prob=0
dataset.required_batch_size_multiple=1
dataset.batch_size=1
```
Do you have any idea for the configurations or about this large WER? 

Thank you again for the great work!

**Comment by JeongHun0716** [comment]
Hi @joyolee,

Thank you for your interest in our work.

I believe there are two potential issues to consider:

Pre-trained Models:

Could you please confirm that the pre-trained model you downloaded is indeed the AV-HuBERT large_vox_iter5.pt?

Also, is it confirmed that the model used is the llama3.2 3B (not instruct)?

Audio-Visual Data Pre-processing:

It seems that the pre-processing of the audio-visual data might differ from my setup.

If you could show me the numpy array (or tensor) from the file test/S2Dj9M71JAc/00006.wav, I can verify whether the audio is the same as mine.

Additionally, to ensure that no noise is added, could you please comment out the code near line 289 in src/dataset.py where it checks for self.subset == 'test' and applies wav_data = self.add_noise(wav_data)?

Also, the code you attached seems to be slightly modified from my version (e.g., batch_size=1). Do you observe the same issue when you run the original script?

**Comment by JeongHun0716** [comment]
100%|███████████████████████████████████████████████████████████████████████████████████████▉| 1320/1321 [22:53<00:00,  1.27it/s]
[2025-04-01 18:48:28,523][hybrid.speech_recognize][INFO] -
INST:Recognize this speech in English. Input :
REF:but i don't
HYP:but i don't

[2025-04-01 18:48:28,523][hybrid.speech_recognize][INFO] - Num_sentences 1,321
[2025-04-01 18:48:28,532][hybrid.speech_recognize][INFO] - WER: 0.717896865520728%

When I ran the code with "dataset.required_batch_size_multiple=1" and "dataset.batch_size=1", I achieved a WER of 0.7178%. Therefore, I believe that issues with preprocessing or with the pre-trained model’s checkpoint could be the reason.

**Comment by matthiasfranz197** [comment]
Hi @JeongHun0716,

I'm working together with @joyolee. Thank you very much for your help and response. 

When we tried to extract the numpy array from the file test/S2Dj9M71JAc/00006.wav, we noticed that the loaded wav_data is of dtype int16. This caused problems in the beginning with the noise addition, so we already had that commented out. But this also led to incorrect features being extracted by Whisper and consequently to the large WER. 
Converting wav_data to dtype float32 solved the problems and now we are able to run inference in clean and noisy conditions and we obtain the same results as you.

Thanks again for your help!
